src/library_manager/lib_manager.c

```diff
@@ -480,6 +473,27 @@ static void lib_manager_update_sof_ctx(void *base_addr, uint32_t lib_id)
 	/* TODO: maybe need to call dcache_writeback here? */
 }
 
+const struct sof_man_module *lib_manager_get_module_manifest(const uint32_t module_id)
+{
+	const uint32_t entry_index = LIB_MANAGER_GET_MODULE_INDEX(module_id);
+	const struct lib_manager_mod_ctx *const ctx = lib_manager_get_mod_ctx(module_id);
+	const struct sof_man_fw_desc *desc;
+
+	if (!ctx || !ctx->base_addr)
+		return NULL;
+
+	desc = (const struct sof_man_fw_desc *)((const char *)ctx->base_addr +
+							      SOF_MAN_ELF_TEXT_OFFSET);
+
```

Comment:
I think you can add here the check

`	if (!desc) {
		tr_err(&lib_manager_tr, "Failed to load manifest");
		return NULL;
	}`

Reply:
@gbernatxintel since `SOF_MAN_ELF_TEXT_OFFSET` is defined as `0x2000` that check would mean that `ctx->base_addr` is `-0x2000` which is rather unlikely

Reply:
OK, thank you for the clarification. I have no further comments